### PR TITLE
T09 create ability to send message

### PR DESCRIPTION
**Pull request by AymenBerb**
- created senMessage function; each time the function is called, a message is sent on the discord channel
- created a time slot button for the tutors to select their time slots

(Sorry for the first PR, I was just testing)

### CODE REVIEW

**Review by Ali-Alkhars — COMMENTED** [review]
When I tried running your code I got this error message immediately.

Screenshot 2022-09-09 at 4 11 06 pm

**Comment by AymenBerb** [comment]
@Ali-Alkhars Make sure that your config.json file includes a "channeID" property with the channel's id as its value.

**Review by Ali-Alkhars — COMMENTED** [review]
There are several things that need to be changed:

### Change 1
This is a minor change just to make things more clear. In the config file change the name "channelID" into "mainChannelId" and do the relevant changes to the index.js file.

### Change 2
The users should get a select menu to choose the timeslot instead of writing their preference because no human will check that. Also, there should be a different select menu for each class if there was more than one class. For example, If one student needed a tutor for both Math GCSE and Computer science, then there should be two select menus instead of one.

### Change 3
Change the embeds into a regular message or figure out a way to disable closing them.

### Change 4
I am aware that you made the bot send the message instantly when it logs in for testing now, but you should remove this after all the reviewers approve your ticket. So should be deleted before merging.

**Review by absirgue — CHANGES_REQUESTED** [review]
Excellent work Aymen!

Fantastic UX and love the fact that you managed to make this neat dropdown work in the end.

I would ask for 3 minor changes:
- Can you make it so that the dropdown requires 2 dates and time to be selected. Not more not less.
- Can you add a 'submit' button to the bottom that will then be used to actually send the request when the tutor is ready to confirm his/her choices
- Can you reformulate a bit the message you send: Change "times per week" to "classes per week" and add a parameter to the function that is "classes duration" (the duration of a single class, can be a 1h session or like 1h30,...)

Thanks a ton! Fantastic work and clean code

**Comment by AymenBerb** [comment]
Class duration should be inputed in the db in minutes please.

Made all required changes.

File: index.js [hunk 1]
```diff
@@ -1,14 +1,115 @@
...
+				.setCustomId('submitButton')
+				.setLabel('Submit request')
+				.setStyle(ButtonStyle.Primary));
+			
+	// Sends both objects to channel
+	channel.send({ embeds: [msgEmbed], components: [row,row2]});
+}
+
+// Stores the submission status
+let answerSubmitted = false;
```
Comment (Ali-Alkhars) [inline index.js]: These shouldn't be global variables for two reasons. 
1- If more than one person fills the select menu simultaneously, the answers will get messed up.
2- only one person can register, then it is blocked for others. I tried it.

File: index.js [hunk 2]
```diff
@@ -1,14 +1,115 @@
...
+
+	if (interaction.customId === 'dateSelection') {
+
+		tutorId = interaction.user.id; 
+		answer = interaction.values; 
+		await interaction.reply({content: "If you're done with your selection, please submit. You can still change your selection.", ephemeral: true})
+
+	}
+
+	if (interaction.customId === 'submitButton') {
```
Comment (Ali-Alkhars) [inline index.js:102]: This part needs comments

**Review by Ali-Alkhars — CHANGES_REQUESTED** [review]
You have done a great job Aymen, amazing spirit! I left two comments on the code that need to be addressed.

**Review by absirgue — APPROVED** [review]
Excellent work Aymen. 
I love the current state of the bot and the handling of issues of when tutors don't enter enough dates is perfect. Congrats!

File: index.js (hunk 1 above)
Comment (AymenBerb) [reply index.js]: Did all changes. Bot works fine now. Multiple person can register.

**Review by Ali-Alkhars — CHANGES_REQUESTED** [review]
Hey Ayman, I know that you have been working hard, so I apologise for being a pain in the ass.

There are issues that have to be addressed. Using global variables creates lots of issues and should be avoided (I'll get to this in a bit). I know that without them it would be difficult or probably impossible to add the submit button, but that's something we could discuss.

We shouldn't use global variables because of the following (these are the only things I identified but could be more):

1. The courseRequestAnswers array will be horribly huge because the bot will always be online and lots of entries will be added, so this might slow down the bot or make it stop working in the future.
2. There will be multiple course requests on the channel, let's say for example there are only two. Each tutor could only submit one interest, then they can't submit an interest for any other message.
3. The globalFrequency will always be the frequency of the last course request. Let's say there was first a course request with frequency 2. After it, another course request was added with frequency 3. If a tutor tried to submit interest for the first request, they will see a message saying they should select 3 choices.
4. If two users submit interest for two different course requests simultaneously, the answer gets mixed up.

You can try these yourself, just let the bot send two different course requests and test with two users.

**Pull request closed without merging**